### PR TITLE
Remove C++ API links from Authentication chapter

### LINKED ISSUE #3505: The Authentication cookbook is full of c++ api links
The [Authentication chapter](https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/authentication.html) in the Cookbook is full of c++ API links instead of the PyQGIS ones

### PR DESCRIPTION
**Pull request by DelazJ**
fix #3505

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 1]
```diff
@@ -253,9 +253,10 @@ enabled service like a WMS or WFS or to a DB connection.
 
   Take into account that not all QGIS data providers are integrated with the
   Authentication infrastructure. Each authentication method, derived from the
-  base class `QgsAuthMethod <https://qgis.org/api/classQgsAuthMethod.html>`_
-  and support a different set of Providers. For example ``Identity-Cert``
-  method supports the following list of providers:
+  base class :class:`QgsAuthMethod <qgis.core.QgsAuthMethod>`
+  and support a different set of Providers. For example :meth:`certIdentity ()
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
  and support a different set of Providers. For example the :meth:`certIdentity ()
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 2]
```diff
@@ -287,11 +288,11 @@ URI parameter with credential just before setting the HTTP connection.
 
 .. warning::
 
-  Developer would have to leave ``authcfg`` expansion to the QgsAuthManager, in
-  this way he will be sure that expansion is not done too early.
+  Developer would have to leave ``authcfg`` expansion to the :class:`QgsAuthManager
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
  The developer would have to leave ``authcfg`` expansion to the :class:`QgsAuthManager
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 3]
```diff
@@ -287,11 +288,11 @@ URI parameter with credential just before setting the HTTP connection.
...
+  <qgis.core.QgsAuthManager>`, in this way he will be sure that expansion is not done too early.
 
-Usually an URI string, build using ``QgsDataSourceURI`` class, is used to set
-QGIS data source in the following way:
+Usually an URI string, build using :class:`QgsDataSourceURI <qgis.core.QgsDataSourceUri>`
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
Usually an URI string, built using :class:`QgsDataSourceURI <qgis.core.QgsDataSourceUri>`
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 4]
```diff
@@ -287,11 +288,11 @@ URI parameter with credential just before setting the HTTP connection.
...
+class, is used to set QGIS data source in the following way:
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
class, is used to set the QGIS data source in the following way:
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 5]
```diff
@@ -382,7 +383,7 @@ Authentication Editor GUI
 
 The complete GUI used to manage credentials, authorities and to access to
 Authentication utilities is managed by the class
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
Authentication utilities is managed by the
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 6]
```diff
@@ -382,7 +383,7 @@ Authentication Editor GUI
...
-`QgsAuthEditorWidgets <https://qgis.org/api/classQgsAuthEditorWidgets.html>`_
+:class:`QgsAuthEditorWidgets <qgis.gui.QgsAuthEditorWidgets>`.
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
:class:`QgsAuthEditorWidgets <qgis.gui.QgsAuthEditorWidgets>` class.
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst (hunk 4 above)
Comment (DelazJ) [reply source/docs/pyqgis_developer_cookbook/authentication.rst]: what's QGIS data source?

File: source/docs/pyqgis_developer_cookbook/authentication.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/pyqgis_developer_cookbook/authentication.rst]: @havatv be careful to not break indentation in your suggestions.

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/authentication.rst]: I try not to. Did I do it here?

File: source/docs/pyqgis_developer_cookbook/authentication.rst (hunk 4 above)
Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/authentication.rst]: Good point.

File: source/docs/pyqgis_developer_cookbook/authentication.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/pyqgis_developer_cookbook/authentication.rst]: yes. see the suggestion before my edit.

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/authentication.rst]: I still fail to see it - sorry. Lack of github knowledge on my side I guess.
And where did the moving of "class" to after the reference go?

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 7]
```diff
@@ -403,7 +404,8 @@ an integrated example can be found in the related `test <https://github.com/qgis
 Authorities Editor GUI
 ----------------------
 
-A GUI used to manage only authorities is managed by the class `QgsAuthAuthoritiesEditor <https://www.qgis.org/api/classQgsAuthAuthoritiesEditor.html>`_
+A GUI used to manage only authorities is managed by the class
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
A GUI used to manage only authorities is managed by the
```

File: source/docs/pyqgis_developer_cookbook/authentication.rst [hunk 8]
```diff
@@ -403,7 +404,8 @@ an integrated example can be found in the related `test <https://github.com/qgis
...
+`QgsAuthAuthoritiesEditor <qgis.gui.QgsAuthAuthoritiesEditor>`.
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/authentication.rst]:
```suggestion
`QgsAuthAuthoritiesEditor <qgis.gui.QgsAuthAuthoritiesEditor>` class.
```

**Comment by DelazJ** [comment]
Thanks @havatv

**Pull request merged**